Allow Normaliz to work with Rational Polytopes

ollieclarke8787 opened the pull request:
Modified the Normaliz package to work with rational coordinates - Macaulay2 can concatenate String | ZZ but cannot concatenate String | QQ

ollieclarke8787 committed:
Update from develop repo

Adding the changes from ollieclarke8787/WorkshopDurham2024@8f94f390933ece5951b0b8995f2393a84ddfdb29 and ollieclarke8787/WorkshopDurham2024@a45e5e9ef2a5646e9dd415447715452372626107

ollieclarke8787 committed:
Merge branch 'Macaulay2:development' into development

ollieclarke8787 committed:
Remove package for in headline

ollieclarke8787 committed:
remove commented code at eof

ollieclarke8787 committed:
remove elapsedTime

ollieclarke8787 committed:
Merge branch 'Macaulay2:development' into development

ollieclarke8787 committed:
Restored keywords

ollieclarke8787 committed:
Merge branch 'development' of github.com:Macaulay2/M2 into normaliz_rp

ollieclarke8787 committed:
allow for rational coordinates when writing data

d-torrance commented:
Thanks!  Would you add a test with a rational polytope that would have failed before the fix?

ollieclarke8787 committed:
update: documentation for writeNmzData

ollieclarke8787 committed:
Add test: normaliz works with rational polytopes

ollieclarke8787 committed:
fix: rational example

ollieclarke8787 commented:
Sure. I've added a non-integral rational example as a test (which doesn't work with the current Normaliz) and changed the documentation in `writeNmzData` to reflect that rational polytopes also work.

d-torrance reviewed (approved):
Thank you!
